pallets/runtime/src/general_tm.rs

```diff
@@ -299,6 +456,110 @@ impl<T: Trait> Module<T> {
+        // ensure whether the trusted issuer's did is register did or not
+        ensure!(
+            <Identity<T>>::is_identity_exists(&trusted_issuer),
+            Error::<T>::DidNotExist
+        );
+        ensure!(
+            !Self::trusted_claim_issuer(&ticker).contains(&trusted_issuer) && is_add_call,
+            Error::<T>::IncorrectOperationOnTrustedIssuer
+        );
+        Self::unsafe_modify_default_trusted_claim_issuer(ticker, trusted_issuer);
+        Ok(())
+    }
+
+    fn modify_default_trusted_claim_issuers_batch(
+        origin: T::Origin,
+        ticker: Ticker,
+        trusted_issuers: Vec<IdentityId>,
+        is_add_call: bool,
+    ) -> DispatchResult {
+        let sender_key = AccountKey::try_from(ensure_signed(origin)?.encode())?;
+        let did = Context::current_identity_or::<Identity<T>>(&sender_key)?;
+        let sender = Signatory::AccountKey(sender_key);
+        ensure!(
+            <Identity<T>>::is_signer_authorized(did, &sender),
+            Error::<T>::SenderMustBeSigningKeyForDid
+        );
+        ensure!(trusted_issuers.len() >= 1, Error::<T>::InvalidLength);
+        ensure!(Self::is_owner(&ticker, did), Error::<T>::Unauthorized);
+        // ensure whether the trusted issuer's did is register did or not
+        for trusted_issuer in trusted_issuers.iter() {
+            match is_add_call {
+                true => ensure!(
+                    !Self::trusted_claim_issuer(&ticker).contains(&trusted_issuer),
+                    Error::<T>::IncorrectOperationOnTrustedIssuer
+                ),
+                false => ensure!(
+                    Self::trusted_claim_issuer(&ticker).contains(&trusted_issuer),
+                    Error::<T>::IncorrectOperationOnTrustedIssuer
+                ),
+            };
+            ensure!(
+                <Identity<T>>::is_identity_exists(trusted_issuer),
+                Error::<T>::DidNotExist
+            );
+        }
+        // iterate all the trusted issuer and modify the data of those.
+        for default_issuer in trusted_issuers.into_iter() {
+            Self::unsafe_modify_default_trusted_claim_issuer(ticker, default_issuer);
+        }
+        Ok(())
+    }
+
+    fn unsafe_change_asset_rule(ticker: Ticker, asset_rule_id: u32, new_asset_rule: AssetRule) {
+        <AssetRulesMap>::mutate(&ticker, |asset_rule| {
+            if let Some(index) = asset_rule
+                .rules
+                .iter()
+                .position(|rule| rule.rule_id == asset_rule_id)
+            {
+                asset_rule.rules[index].asset_rule = new_asset_rule.clone();
+            }
+        });
+        Self::deposit_event(Event::ChangeAssetRule(
+            ticker,
+            asset_rule_id,
+            new_asset_rule,
+        ));
+    }
+
+    fn get_next_rule_id(ticker: Ticker) -> u32 {
+        u32::try_from(Self::asset_rules(ticker).rules.len()).unwrap_or_default()
```

Comment:
This will cause issues when rules are removed. Let's say there are rules A, B, and C right now with rule_id 0, 1, and 2. if rule B is deleted, we are left with A, C and id 0, 2. Now, if a new rule, D is added, we will have A, C, and D with id 0, 2, 2 which is not right.